.github/workflows/odh-create-release-tag.yaml

```
@@ -0,0 +1,125 @@
+name: Create Tag and Release with Changelog
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag_name:
+        description: 'New release tag (e.g., odh-v2.35)'
+        required: true
+        type: string
+      next_odh_tag:
+        description: 'Next development tag (e.g., odh-v2.36)'
+        required: true
+        type: string
+    # Note: This workflow assumes the release tag and image tags are in sync.
+    #       'tag_name' is used to create the release and as the value to find in files.
+    #       'next_odh_tag' provides the new value to set.
+
+permissions:
+  contents: write
+  packages: write
+  pull-requests: write
+  
+jobs:
+  fetch-tag:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
+          fetch-depth: 0
+      
+      - name: print tag
+        id: print_tag
+        run: | 
+          echo "NEW_TAG=${{ github.event.inputs.tag_name }}" >> $GITHUB_ENV
+          echo "$(basename ${{ github.ref }})"
+
+      - name: Check if tag exists
+        id: check_tag
+        run: |
+         import sys
+         import subprocess
+         tag_name = "${{ github.event.inputs.tag_name }}"
+         command = ['git', 'tag', '-l', tag_name]
+         output = subprocess.check_output(command, stderr=subprocess.STDOUT)
+         if output.decode() != "":
+           print(f"Error: Tag '{tag_name}' already exists.", file=sys.stderr)
+           sys.exit(1)
+         else:
+           print(f"Tag '{tag_name}' does not exists.")
```

Nit **Typo on line 51**: "does not exists" to "does not exist".

```suggestion
           print(f"Tag '{tag_name}' does not exist.")
```